### ISSUE TITLE
Verification Info: As an Industry Reporter, I want to fill out the LFO verification form

### ISSUE THREAD

**Issue by shon-button** [bcgov/cas-reporting#410]
#### Description:

All Regulated Operations must fill out this page. 
Reporting Operations who have emissions above 25,000 T CO2e will also need to do this step (reporting operations under 25,000T will not). 

[LFO Wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=17438-4851&t=kkLCs25GtG7LwPro-4)

#### Acceptance Criteria:

Given I am a REPORTING Operation who has attributable emissions greater than or equal to 25,000 T CO2e (not including CO2 from biomass listed in item 1 of Schedule C)
Then I need to fill out the verification LFO form page. 

Given that I am on the Sites visited dropdown,
When I select "Other" from the dropdown, 
Then I can still select other sites, i.e., Site 1, Other, Site 2

Given that I select "Other" from the dropdown, 
The "Other" form fields appear, and a button appears that says "Add another site"
Given that I click "Add another site" another "Other" form field appears.
<img width="689" alt="Image" src="https://github.com/user-attachments/assets/191d4c7f-facd-4a94-8c76-30e41899cb42" />

#### Development Checklist:

- [x] In `bc_obps/reporting/models/report_verification.py` remove information about a report verification visit
- [x] In `bc_obps/reporting/schema/report_verification.py` remove information about a report verification visit
- [x] Create model ReportVerificationVisit for information about a report verification visit in `bc_obps/reporting/models/report_verification_visit.py` 
- [x] In `bc_obps/reporting/models/__init__.py` add references for `ReportVerificationVisit`
- [ ] Create model migration 
 ```
cd bc_obps
make migrations
make reset_db
```
- [x] In `ReportVerification` tests remove information about a report verification visit
- [x] In `bc_obps/registration/tests/models/test_user.py` add reference for ReportVerificationVisit TimeStampedModel
- [x] In `bc_obps/reporting/tests/models/test_report_verification.py` add foreign key relationship 
            ("reportverificationvisit_records", "report verification visit", None, 0),

- [x] Update frontend schemas to manage array of sites visited; array of types of site visited `bciers/apps/reporting/src/data/jsonSchema/signOff/verification/verification.ts`
- [x] Update backend model to accept array of sites visited in file `bc_obps/reporting/models/report_verification.py`
- [x] Update backend schema to accept array of sites visited in file `bc_obps/reporting/schema/report_verification.py`
- [ ] Update service to save array of sites visited; array of types of site visited `bc_obps/reporting/service/report_verification_service.py`
- [x] Update vitests as needed: `bciers/apps/reporting/src/tests/components/signOff/verification/VerificationForm.test.tsx`
- [ ] Update tests as needed: `bc_obps/reporting/tests/api/test_report_verification_api.py`
- [ ] Update tests as needed: `bc_obps/reporting/tests/service/test_report_verification_service.py`
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

- Business Area creating a template for this so we're blocked on this page. 
bcgov/cas-registration#200 blocked by this

**Comment by patriciarussellCAS** [comment]
This ticket already exists in the reporting repo

**Comment by shon-button** [comment]
@patriciarussellCAS 

This ticket is specific to LFO; so, a split from the [existing ticket](https://github.com/bcgov/cas-reporting/issues/199)...as such, I have re-opened the ticket.

**Comment by zoeyli-46** [comment]
@shon-button could I add the following to the Acceptance Criteria:
If the user selects "Other" They can still select other facilities. i.e., Facility 1, Other

**Comment by shon-button** [comment]
@patriciarussellCAS 

Reachable at:

#### Regression test: SFO "Site(s) Visited" conditional fields*

1. Navigate to https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/
2. Click button "Log in with Business BCeID"
3. Navigate to  `Reports\Bugle SFO - Registered` (https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports) verification step: ./verification
4. Complete the form with field value `Site(s) Visited`= **"None"**
5. Click button `Save & Continue`
**Expected Results**
- Form data is saved to the database
- `Attachment` page is displayed
6. Navigate to `Bugle SFO - Registered` verification step: ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/d47d0699-39af-4617-a749-8bf6c8e7dc34)
7. Complete the form with field value `Site(s) Visited`= **"Other"**
8. Click button `Save & Continue`**Expected Results**
- Form data is saved to the database
- `Attachment` page is displayed
9. Navigate to `Bugle SFO - Registered` verification step: ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/67cf6688-a181-4498-99e1-47e8b797e4f8)
10. Complete the form with field value `Site(s) Visited`= **"Facility 22"**
11. Click button `Save & Continue`
**Expected Results**
- Form data is saved to the database
- `Attachment` page is displayed
12. Navigate to `Bugle SFO - Registered` verification step:  ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/f715cf07-ddc6-49ab-a4c4-e8056cd10159)




#### QA Test- LFO "Site(s) Visited" conditional fields*

1. Navigate to http://localhost:3000
2. Click button "Log in with Business BCeID"
3. Navigate to `Reports\Banana LFO - Registered` (https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports)  verification step: ./verification
4. Complete the form with field value `Site(s) Visited`= **"None"**
5. Click button `Save & Continue`
**Expected Results**
- Form data  is saved to the database
- `Attachment` page is displayed
6. Navigate to `Banana LFO - Registered`  verification step: ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/b45aa20b-b4d9-484e-83be-a9f6d3ae62db)
7. Complete the form with field value `Site(s) Visited`= **"Other"**
7.1 Complete multiple "Add Other Visit"
8. Click button `Save & Continue`
**Expected Results**
- Form data  is saved to the database
- `Attachment` page is displayed
9. Navigate to ``Banana LFO - Registered` ` verification step: ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/04c99fb0-6983-4edc-8f62-340c4fdc2098)
10. Complete the form with field value `Site(s) Visited`= **"Facility 1"**
11. Click button `Save & Continue`
**Expected Results**
- Form data is saved to the database
- `Attachment` page is displayed
12. Navigate to `Banana LFO - Registered` verification step: ./verification
**Expected Results**
- Form data is displayed from the database
![image](https://github.com/user-attachments/assets/236e4603-8502-4cf3-a71c-834eb3235dae)

**Optional:** Test using different `Visit Names` combinations

![Image](https://github.com/user-attachments/assets/82663430-3d2d-454d-a68f-1e201e698858)

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EQgC33G82BFOpIiN0qUwTlcBgfwkmgA4JWFZSV1eHPgbsA?e=RJUlW0).

I can verify that:
- [x] When filling out an SFO/LFO Verification form and I click on ```Save & Continue``` button, the page redirects to the Attachments page
- [x] From the Attachments page, when I click the ```Back``` button, it redirects to the Verification page and it shows the latest data I saved in the Verification page
- [x] When filling out an LFO report and I choose **Other** for the value of ```Site(s) visited```, then the ```Add Other Visit``` button is displayed
- [x] When I click on the ```Add Other Visit``` button, three new required fields (```Name```, ```Coordinates```, and ```Visit Type```) are displayed
- [x] After clicking ```Add Other Visit```, I am able to see a button that will allow me to delete the entry for "Other visit"
- [x] I am able to mix **Other**  and **Facility X** in the ```Site(s) visited``` field
- [x] When I have data already added for **Other** and I add **Facility X** in the ```Site(s) visited```, the previous data that I added for **Other** is not deleted, rather a new set of radio buttons are presented where I can choose if ```Virtual``` or ```In person``` for **Facility X**

Moving to Pending PO for review of notes and test. Thank you!

**Comment by patriciarussellCAS** [comment]
Test document reviewed. Thanks for a thorough QA. Moving to Accepted.
